pkgs/applications/networking/jdownloader2/default.nix

```
@@ -0,0 +1,71 @@
+{ stdenv, lib, fetchurl, writeScript, makeDesktopItem, jre, imagemagick, coreutils }:
+
+let
+  jddesktop = makeDesktopItem rec {
+    name = "JDownloader 2";
+    exec = wrapper;
+    icon = "jdownloader";
+    comment = "Free, open-source download management tool.";
+    desktopName = "JDownloader 2";
+    genericName = "JDownloader 2";
+    categories = "Network;";
+  };
+
+  wrapper = writeScript "jdownloader2.sh" ''
+    #! ${stdenv.shell}
+    PATH=${lib.makeBinPath [ jre coreutils ]}
+
+    JDDIR=''${XDG_DATA_HOME:-$HOME/.local/share}/jdownloader
+    JDJAR=''${JDDIR}/JDownloader.jar
+    dir=`dirname "$0"`
+
+    if [ ! -f ''${JDJAR} ]; then
+        install -Dm755 $dir/JDownloader.jar ''${JDJAR}
+    fi
+
+    ${jre}/bin/java -jar ''${JDDIR}/JDownloader.jar "''${@}"
+  '';
+
+in stdenv.mkDerivation rec {
+  name = "jdownloader2";
+  version = "2.0";
+
+  src = fetchurl {
+    url = "http://installer.jdownloader.org/JDownloader.jar";
+    sha256 = "9951b786e24fc3777a0df0a7b516ba53d0c8e778d6a69ebc29dcff86ee6b5829";
+  };
+  dontUnpack = true;
+
+  icon = fetchurl {
+    url = "https://jdownloader.org/_media/vote/trazo.png";
+    sha256 = "3ebab992e7dd04ffcb6c30fee1a7e2b43f3537cb2b22124b30325d25bffdac29";
+  };
+
+
+
+  nativeBuildInputs = [ imagemagick ];
+
+  installPhase = ''
+  mkdir -pv $out/bin $out/share/applications
+  cp ${src} $out/bin/JDownloader.jar
+  cp ${wrapper} "$out/bin/jdownloader.sh"
+  cp "${jddesktop}/share/applications/JDownloader 2.desktop" "$out/share/applications/JDownloader 2.desktop"
+
+  # create icons
+  for size in 16 32 48 64 72 96 128 192 512 1024; do
+    mkdir -p $out/share/icons/hicolor/"$size"x"$size"/apps
+    convert -resize "$size"x"$size" \
+      ${icon} \
+      $out/share/icons/hicolor/"$size"x"$size"/apps/jdownloader.png
+  done
```

Nit: will be nice to add more indent here.
```suggestion
    mkdir -pv $out/bin $out/share/applications
    cp ${src} $out/bin/JDownloader.jar
    cp ${wrapper} "$out/bin/jdownloader.sh"
    cp "${jddesktop}/share/applications/JDownloader 2.desktop" "$out/share/applications/JDownloader 2.desktop"

    # create icons
    for size in 16 32 48 64 72 96 128 192 512 1024; do
      mkdir -p $out/share/icons/hicolor/"$size"x"$size"/apps
      convert -resize "$size"x"$size" \
        ${icon} \
        $out/share/icons/hicolor/"$size"x"$size"/apps/jdownloader.png
    done
```